trie-db/src/triedb.rs

```diff
@@ -451,42 +439,39 @@ impl<'a, 'cache, L: TrieLayout> TrieDBIterator<'a, 'cache, L> {
 		prefix: &[u8],
 		start_at: &[u8],
 	) -> Result<Self, TrieHash<L>, CError<L>> {
-		let mut inner = TrieDBNodeIterator::new(db)?;
-		inner.prefix_then_seek(prefix, start_at)?;
+		Ok(Self { db, raw_iter: TrieDBRawIterator::new_prefixed_then_seek(db, prefix, start_at)? })
+	}
 
-		Ok(TrieDBIterator { inner })
+	/// Restore an iterator from a raw iterator.
+	pub fn from_raw(db: &'a TrieDB<'a, 'cache, L>, raw_iter: TrieDBRawIterator<L>) -> Self {
+		Self { db, raw_iter }
+	}
+
+	/// Convert the iterator to a raw iterator.
+	pub fn into_raw(self) -> TrieDBRawIterator<L> {
+		self.raw_iter
 	}
 }
 
 impl<'a, 'cache, L: TrieLayout> TrieIterator<L> for TrieDBIterator<'a, 'cache, L> {
 	/// Position the iterator on the first element with key >= `key`
 	fn seek(&mut self, key: &[u8]) -> Result<(), TrieHash<L>, CError<L>> {
-		TrieIterator::seek(&mut self.inner, key)
+		self.raw_iter.seek_prefix(self.db, key).map(|_| ())
```

Comment:
I am wondering if the function for raw_iter `seek_prefix` should be rename (I am maybe the one that give this ~ naming in the first place ). Here it just does a seek key which is what we need, but I had to check the code to be sure it is not the variant that seek a key and then remove nodes from the stack so it limits remaining iteration to the prefix.